client/client.go

```diff
@@ -538,3 +542,32 @@ func (c *Client) Collect(ch chan<- prometheus.Metric) {
 var (
 	_ prometheus.Collector = (*Client)(nil)
 )
+
+// argListFromMySqlParams creates an array of strings from the pointer to the parameters for pt-mysql-sumamry
+func argListFromMySqlParams(pParams *agentpb.StartActionRequest_PTMySQLSummaryParams) []string {
+	var args []string
+
+	// Only adds the arguments are valid
+	// If socket valid, socket will be used. Otherwise it will try user and password.
+	if pParams.Socket != "" {
+		args = append(args, "--socket="+pParams.Socket)
+	} else {
+		if pParams.Username != "" {
+			args = append(args, "--user="+pParams.Username)
+		}
```

Comment:
Socket or Host + port.
User and pass can be used with socket and host.

Reply:
This requirement is not clear. OR or XOR? And brackets would be helpful too...

Reply:
pt-mysql-summary accepts any combination of socket/host/port/user/pass https://github.com/percona/percona-toolkit/blob/3.x/bin/pt-mysql-summary#L482-L503